diff --git a/include/fmt/format.h b/include/fmt/format.h
index 7637c8a0..d750043f 100644
--- a/include/fmt/format.h
+++ b/include/fmt/format.h
@@ -1332,7 +1332,8 @@ template <typename Char, typename UInt, typename Iterator,
 FMT_CONSTEXPR inline auto format_decimal(Iterator out, UInt value, int size)
     -> format_decimal_result<Iterator> {
   // Buffer is large enough to hold all digits (digits10 + 1).
-  Char buffer[digits10<UInt>() + 1] = {};
+  constexpr int ndigits = digits10<UInt>() + 1;
+  Char buffer[ndigits] = {};
   auto end = format_decimal(buffer, value, size).end;
   return {out, detail::copy_str_noinline<Char>(buffer, end, out)};
 }
